app/controllers/books_controller.rb

```diff
@@ -0,0 +1,29 @@
+class BooksController < ApplicationController
+  def show
+    @book = Book.find_by id: params[:id]
```

Comment:
Nếu id nil thì sao em?

Reply:
Theo em, show này là khi có book người dùng click vào mới chạy ra show thì đâu có trường hợp nill ạ.

Reply:
nếu anh gõ thẳng ở trên URL thì sao :-?

Reply:
Dạ vâng, e hiểu rồi ạ. 
Vậy e nghĩ sẽ dùng   if params[:id].present? và in ra 1 câu thông báo có ổn không ạ,

Reply:
đc hoặc dùng find thì nó sẽ trả về exception đó

Reply:
Dạ vâng, e hiểu rồi ạ.